__tests__/gui/tabular_comparison/tabular_comparison.js

```diff
@@ -0,0 +1,87 @@
+import {When, Given, Then} from "cypress-cucumber-preprocessor/steps";
+import {
+    gotoTabularComparison,
+    setupInterceptions,
+    setupInterceptionsForSpecificGeo
+} from "../common_cy_functions/general";
+import profiles from "./profiles.json";
+import profile from "./profile.json";
+import geographies from './geographies.json';
+import all_details_ec from "./all_details_ec.json";
+import all_details_wc from "./all_details_wc.json";
+
+
+Given('I am on the tabular comparison view', () => {
+    setupInterceptions(profiles, null, profile, null, null);
+    setupInterceptionsForSpecificGeo("EC", all_details_ec);
+    setupInterceptionsForSpecificGeo("WC", all_details_wc);
+    gotoTabularComparison();
+})
+
+Then('I wait until view is ready', () => {
+    cy.get('[data-testid=comparison-geographies]', {timeout: 20000}).should('be.visible');
+})
+
+When(/^I click on (\w+) autocomplete$/, (type) => {
+   cy.get(`[data-testid=${type}-autocomplete]`).last().find('input').click();
+})
+
+Then('I assert no options in dropdown', () => {
+  cy.get('.MuiAutocomplete-popper').contains('No options');
+
+})
+
+Then(/^I search for (\w+) in (\w+) autocomplete$/, (value, type) => {
+  if (type === "geography"){
+    cy.intercept(`/api/v1/geography/search/8/*`, (req) => {
+        req.reply({
+            statusCode: 200,
+            body: geographies.filter(obj => obj.name.toLowerCase().includes(value)),
+            forceNetworkError: false // default
+        })
+    }).as('selectedgeo');
+  }
+
+  cy.get(`[data-testid=${type}-autocomplete]`).last().find('input').type(value);
+})
+
+Then(/^I select ([\w ]+) in autocomplete dropdown$/, (value) => {
+  cy.wait(1000);
```

Comment:
Does cypress have a 'wait until' feature? If possible it would be better to wait until the element contains what we want to test and proceed as soon as it does

Reply:
There is a separate package for this : https://www.npmjs.com/package/cypress-wait-until
Should we include it ?
It is only required when we remove value from autocomplete and re add it
May be we should do it when we are writing extensive tests for tabular comparison 🤔

Reply:
Yes we can wait for more extensive tests or if it causes a race condition

Reply:
why do we even need ` cy.wait(1000);` here? 

`cy.get(".MuiAutocomplete-popper li[data-option-index='0']", {timeout: 20000})`
this code will already wait max 20 seconds until it can find `.MuiAutocomplete-popper li[data-option-index='0` already. no need for an extra cy.wait or extra packages.

have you tested it without cy.wait @goyal1092 ? it should work

Reply:
yup I have tested without wait @emre2038 
timeout is only checks if element exist and what we are waiting for is to update results in dropdown.
We will have to use wait until value of 0 matches the value.
Also we require Wait in only one condition that is searching for WC instead of EC.

Reply:
proposed :

```
Then(/^I select ([\w ]+) in autocomplete dropdown$/, (value) => {
  cy.get(`.MuiAutocomplete-popper li[data-option-index='0']:contains(${value})`, {timeout: 20000}).click()
})
```